app/src/main/java/com/android/bookswap/ui/chat/ChatScreen.kt

```diff
@@ -173,6 +175,56 @@ fun ChatScreen(
         }
       }
       delay(2000) // Delay for 2 seconds
+
+      // Check and add contacts for both users
+      if (messages.isNotEmpty()) {
+        // Add otherUser to currentUser's contacts
+        userSource.getUser(currentUser.userUUID) { result ->
+          if (result.isSuccess) {
+            val updatedUser = result.getOrThrow()
+            if (!updatedUser.contactList.contains(otherUser.userUUID.toString())) {
+              userSource.addContact(currentUser.userUUID, otherUser.userUUID.toString()) {
+                  contactResult ->
+                if (contactResult.isSuccess) {
+                  Log.d(
+                      "ChatScreen",
+                      "Added ${otherUser.userUUID} to ${currentUser.userUUID}'s contacts")
+                } else {
+                  Log.e(
+                      "ChatScreen",
+                      "Failed to add contact: ${contactResult.exceptionOrNull()?.message}")
+                }
+              }
+            }
+          } else {
+            Log.e(
+                "ChatScreen", "Failed to fetch current user: ${result.exceptionOrNull()?.message}")
+          }
+        }
+
+        // Add currentUser to otherUser's contacts
+        userSource.getUser(otherUser.userUUID) { result ->
+          if (result.isSuccess) {
+            val updatedOtherUser = result.getOrThrow()
+            if (!updatedOtherUser.contactList.contains(currentUser.userUUID.toString())) {
+              userSource.addContact(otherUser.userUUID, currentUser.userUUID.toString()) {
+                  contactResult ->
+                if (contactResult.isSuccess) {
+                  Log.d(
+                      "ChatScreen",
+                      "Added ${currentUser.userUUID} to ${otherUser.userUUID}'s contacts")
+                } else {
+                  Log.e(
+                      "ChatScreen",
+                      "Failed to add contact: ${contactResult.exceptionOrNull()?.message}")
+                }
+              }
+            }
+          } else {
+            Log.e("ChatScreen", "Failed to fetch other user: ${result.exceptionOrNull()?.message}")
+          }
+        }

```

Comment:
This implementation doesnt comply with the MVVM template as this is calling data from the UI and should be changed as soon as possible.

Reply:
same as last time